ui/plugins/pluginpanel.html

```diff
@@ -245,15 +241,17 @@
         <ul class="nav navbar-nav info-panel-navbar-controls">
           <li>
             <a href="" ng-click="togglePinboardVisibility()"
-               ng-class="{'pinboard-open-btn': isPinboardVisible}">
-              <span ng-hide="isPinboardVisible"
-                    title="Open the pinboard">Open pinboard&nbsp
-                    <span class="fa fa-angle-up"></span>
-              </span>
-              <span ng-show="isPinboardVisible"
-                    title="Close the pinboard">Close pinboard&nbsp
-                    <span class="fa fa-angle-down"></span>
-              </span>
+               class="pinboard-btn">
+              <div ng-attr-title="{{isPinboardVisible ? 'Close the pinboard' : 'Open the pinboard'}}">Pinboard
+                <div ng-if="pinboard_service.count.numPinnedJobs"
+                     class="pin-count-group"
+                     ng-class="{'pin-count-group-3-digit': (pinboard_service.count.numPinnedJobs > 99)}">
+                  <div class="pin-count-text"
+                       ng-class="{'pin-count-text-3-digit': (pinboard_service.count.numPinnedJobs > 99)}">
+                    {{getCountPinnedJobs()}}</div>
+                </div>
+                <span class="fa" ng-class="isPinboardVisible ? 'fa-angle-down' : 'fa-angle-up'"></span>
+              </div>
```

Comment:
We turn the duplicate close/open markup blocks, into one here. And in edge pinning cases where we go to 3 digits (>99 <=500 pinned) we increase the count group size slightly to accommodate the extra digit.